feature-libs/quote/components/items/quote-items.component.service.ts

```diff
@@ -4,24 +4,136 @@
-  public setQuoteEntriesExpanded(expanded: boolean): void {
+  /**
+   * Signals that the quote entry section is expanded.
+   *
+   * @param expanded Quote entry section expanded?
+   */
+  setQuoteEntriesExpanded(expanded: boolean): void {
     this._quoteEntriesExpanded$.next(expanded);
   }
-
-  public getQuoteEntriesExpanded(): Observable<boolean> {
+  /**
+   * Emits expanded state of the quote entries section.
+   *
+   * @returns Observable of boolean values telling if the quote entry section is expanded
+   */
+  getQuoteEntriesExpanded(): Observable<boolean> {
     return this._quoteEntriesExpanded$;
   }
+
+  /**
+   * Retrieves observable of current quote entries (which can be read from a quote,
+   * a saved cart or an active cart, depending on the quote state and its edit status).
+   *
+   * @returns Observable of quote items data. The return structure also contains information
+   * about the type of the hosting document (quote, saved cart or active cart) and its edit status.
+   */
+  retrieveQuoteEntries(): Observable<QuoteItemsData> {
+    return this.userIdService.takeUserId().pipe(
+      switchMap((userId) =>
+        combineLatest([this.quoteFacade.getQuoteDetails(), of(userId)])
+      ),
+      tap(([quote, userId]) => {
+        this.loadQuoteCartIfNeeded(quote, userId);
+      }),
+      switchMap(([quote, _userId]) => {
+        if (!quote.cartId) {
+          return this.prepareQuote(quote);
+        } else {
+          const quoteCartId: string = quote.cartId;
+          if (!quote.isEditable) {
+            return this.prepareSavedCart(quoteCartId);
+          } else {
+            return this.prepareActiveCart();
+          }
+        }
+      }),
+      filter(([abstractOrder, _editState]) => abstractOrder !== undefined),
+      map(([abstractOrder, editState, abstractOrderType]) => {
+        return {
+          entries: abstractOrder.entries,
+          readOnly: editState.readOnly,
+          abstractOrderId: abstractOrder.code,
+          abstractOrderType: abstractOrderType,
+        };
+      })
+    );
+  }
+
+  protected prepareActiveCart(): QuoteOrCartWithEditState {
+    return combineLatest([
+      this.activeCartFacade.isStable().pipe(
+        filter((stable) => stable),
+        switchMap(() => this.activeCartFacade.getActive())
+      ),
+      of({ readOnly: false }),
+      of(AbstractOrderType.CART),
+    ]);
+  }
+
+  protected prepareSavedCart(quoteCartId: string): QuoteOrCartWithEditState {
+    return combineLatest([
+      this.multiCartFacade.isStable(quoteCartId).pipe(
+        filter((stable) => stable),
+        switchMap(() => this.multiCartFacade.getCart(quoteCartId))
+      ),
+      of({ readOnly: true }),
+      of(AbstractOrderType.SAVED_CART),
+    ]);
+  }
+
+  protected prepareQuote(quote: Quote): QuoteOrCartWithEditState {
+    return zip(of(quote), of({ readOnly: true }), of(AbstractOrderType.QUOTE));
+  }
+
+  protected loadQuoteCartIfNeeded(quote: Quote, userId: string) {
```

Comment:
I would suggest to rename the method to `loadQuoteCart()`.
According to my point of view it is not required to append  `IfNeed` at the end of the method.

Reply:
But the method first checks if it's needed to load the cart, why should that _not_ be reflected in its name?